fix: ordinals-p2p.tools.ts build error — missing @noble/hashes type declarations

## Problem

`npm run build` fails with two type errors in `src/tools/ordinals-p2p.tools.ts`:

```
src/tools/ordinals-p2p.tools.ts(23,24): error TS2307: Cannot find module '@noble/hashes/sha2.js' or its corresponding type declarations.
src/tools/ordinals-p2p.tools.ts(24,28): error TS2307: Cannot find module '@noble/hashes/utils.js' or its corresponding type declarations.
```

These are the only source-level build errors in the project (other errors are node_modules type noise).

## Likely Cause

The `.js` extension in the import paths may not resolve correctly with the current `moduleResolution` setting, or `@noble/hashes` may need to be added as a direct dependency (currently likely a transitive dep).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

--- a/src/tools/news.tools.ts
+++ b/src/tools/news.tools.ts
@@ -396,7 +396,7 @@ Fields:
         if (body) {
           payload.body = body;
         }
-        if (disclosure) {
+        if (disclosure !== undefined) {
           payload.disclosure = disclosure;
         }
 

